src/System.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/PoolBlockPeriodTest.netcoreapp.cs

```diff
@@ -0,0 +1,161 @@
+            SqlConnection.ClearAllPools();
+            string serverName = Params[0] as string;
+            PoolBlockingPeriod? policy = null;
+
+            if (Params.Length > 1)
+            {
+                policy = (PoolBlockingPeriod)Params[1];
+            }
+
+            string connString = CreateConnectionString(serverName, policy);
+            PoolBlockingPeriodNonAzureTest(connString, policy);
+        }
+
+        [Theory]
+        [InlineData("Test policy with Auto (lowercase)", new object[] { "auto" })]
+        [InlineData("Test policy with Auto (miXedcase)", new object[] { "auTo" })]
+        [InlineData("Test policy with Auto (Pascalcase)", new object[] { "Auto" })]
+        [InlineData("Test policy with Always (lowercase)", new object[] { "alwaysblock" })]
+        [InlineData("Test policy with Always (miXedcase)", new object[] { "aLwAysBlock" })]
+        [InlineData("Test policy with Always (Pascalcase)", new object[] { "AlwaysBlock" })]
+        [InlineData("Test policy with Never (lowercase)", new object[] { "neverblock" })]
+        [InlineData("Test policy with Never (miXedcase)", new object[] { "neVeRblock" })]
+        [InlineData("Test policy with Never (Pascalcase)", new object[] { "NeverBlock" })]
+        public void TestSetPolicyWithVariations(string description, object[] Params)
+        {
+            SqlConnection.ClearAllPools();
+            string policyString = Params[0] as string;
+            PoolBlockingPeriod? policy = null;
+            if (policyString.ToLower().Contains("auto"))
+            {
+                policy = PoolBlockingPeriod.Auto;
+            }
+            else if (policyString.ToLower().Contains("always"))
+            {
+                policy = PoolBlockingPeriod.AlwaysBlock;
+            }
+            else
+            {
+                policy = PoolBlockingPeriod.NeverBlock;
+            }
+            string connString = CreateConnectionString(_sampleAzureEndpoint, null) + $";{_policyKeyword}={policyString}";
+            PoolBlockingPeriodAzureTest(connString, policy);
+        }
+
+        public void PoolBlockingPeriodNonAzureTest(string connStr, PoolBlockingPeriod? policy)
+        {
+            SqlConnection.ClearAllPools();
+            int firstErrorTimeInSecs = GetConnectionOpenTimeInSeconds(connStr);
+            int secondErrorTimeInSecs = GetConnectionOpenTimeInSeconds(connStr);
+            switch (policy)
+            {
+                case PoolBlockingPeriod.Auto:
+                case PoolBlockingPeriod.AlwaysBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 0, firstErrorTimeInSecs);
+                    break;
+                case PoolBlockingPeriod.NeverBlock:
+                    Assert.InRange(secondErrorTimeInSecs, 1, int.MaxValue);
+                    break;
+            }
+        }
+
+        public void PoolBlockingPeriodAzureTest(string connStr, PoolBlockingPeriod? policy)
+        {
+            SqlConnection.ClearAllPools();
```

Comment:
I am curious, why are you clearing all pools? Is it possible to avoid this API usage. In case this is being done, so that the pool is reset for the connection string, then you could have a workaround like providing a randomly generated `Application Name` in the connection string which will lead to the utilization of a new pool.